bin/src/lib.rs

```diff
@@ -741,6 +763,119 @@ impl TlsCertConfig {
     }
 }
 
+/// Persistence configuration for data unrelated to zone data.
+#[derive(Debug, Default, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct StatePersistence {
+    /// Recursive resolver opportunistic encryption state persistence configuration
+    #[cfg(feature = "recursor")]
+    #[serde(default)]
+    pub opportunistic_encryption: Option<OpportunisticEncryptionPersistence>,
+}
+
+/// Configuration for loading/saving per-nameserver encrypted transport protocol state
+/// probe data when opportunistic encryption is enabled for a recursive resolver external
+/// zone.
+#[cfg(feature = "recursor")]
+#[derive(Debug, Deserialize)]
+#[serde(tag = "type", rename_all = "lowercase", deny_unknown_fields)]
+pub enum OpportunisticEncryptionPersistence {
+    /// TOML file state persistence
+    TomlFile {
+        /// Path to the TOML state file used for saving/loading opportunistic encryption state
+        ///
+        /// This file will be loaded at startup, and written to every `save_interval`.
+        file_path: PathBuf,
+        /// Interval in seconds for periodic state saving (0 to disable)
```

Comment:
Is there any useful scenario in which it makes sense to retrieve but not save? If we're going to enable persistence without saving I'd suggest wrapping the `Duration` in an `Option` rather than relying on a sentinel value.